docs/topics/compatibility-guides/compatibility-guide-20.md

```diff
@@ -54,57 +54,529 @@ perspective
+
+### Change inferred type of prefix increment to match getter's return type instead of inc() operator's return type
+
+> **Issue**: [KT-57178](https://youtrack.jetbrains.com/issue/KT-57178)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Deprecation cycle**:
+>
+> - 2.0.0: implement new behavior; inferred type of prefix increment is changed to match getter's return type instead of
+>   the `inc()` operator's return type
+
+### Enforce bound checks when inheriting inner classes from generic inner classes declared in superclasses
+
+> **Issue**: [KT-61749](https://youtrack.jetbrains.com/issue/KT-61749)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Deprecation cycle**:
+>
+> - 2.0.0: report an error when upper bound of the type parameter of a generic inner superclass is violated
+
+### Forbid assigning callable references with SAM types when the expected type is a function type with a function type parameter
+
+> **Issue**: [KT-64342](https://youtrack.jetbrains.com/issue/KT-64342)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Deprecation cycle**:
+>
+> - 2.0.0: report a compilation error on callable references with SAM types when the expected type is a function type
+>   with a function type parameter
+
+### Consider companion object scope for annotation resolution on companion objects
+
+> **Issue**: [KT-64299](https://youtrack.jetbrains.com/issue/KT-64299)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: behavioral
+>
+> **Deprecation cycle**:
+>
+> - 2.0.0: implement new behavior; the companion object scope is now not ignored during annotation resolution on companion objects
+
+### Change evaluation semantics for combination of safe calls and convention operators
+
+> **Issue**: [KT-41034](https://youtrack.jetbrains.com/issue/KT-41034)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: behavioral
+>
+> **Deprecation cycle**:
+>
+> - 1.4.0: report warnings on each incorrect call
+> - 2.0.0: implement new resolution behavior
+
+### Require properties with backing field and a custom setter to be immediately initialized
+
+> **Issue**: [KT-58589](https://youtrack.jetbrains.com/issue/KT-58589)
+> 
+> **Component**: Core language
+>
+> **Incompatible change type**: behavioral
+>
+> **Deprecation cycle**:
+>
+> - 1.9.20: introduce thr `MUST_BE_INITIALIZED` warning for cases where there is no primary constructor
+> - 2.0.0: raise the warning to an error
+
+### Prohibit Unit conversion on arbitrary expressions in invoke operator convention call
+
+> **Issue**: [KT-61182](https://youtrack.jetbrains.com/issue/KT-61182)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: behavioral
+>
+> **Deprecation cycle**:
+>
+> - 2.0.0: report to an error when Unit conversion is applied to an arbitrary expression on variable and invoke resolution
```

Comment:
Could you please also mention a workaround I mentioned in the migration plan in the issue?

Reply:
Added! Can you check and let me know if it's OK?